test/e2e/main_test.go

```diff
@@ -208,109 +240,56 @@ func testInstall(
 	meteringOperatorImageTag,
 	testName string,
 	targetPods int,
+	testInstallFunctions []InstallTestCase,
 	spec metering.MeteringConfigSpec,
 ) {
-	// create a directory used to store the @testName container and resource logs
-	testOutputDir := filepath.Join(df.LoggingPath, testName)
-	err := os.Mkdir(testOutputDir, 0777)
-	assert.NoError(t, err, "creating the test case output directory should produce no error")
-
-	// randomize the namespace to avoid existing namespaces
-	rand.Seed(time.Now().UnixNano())
-	namespace := df.NamespacePrefix + "-" + strconv.Itoa(rand.Intn(50))
+	// create a @testName base directory for the @testInstallFunctions output
+	testCaseOutputBaseDir := filepath.Join(df.LoggingPath, testName)
+	err := os.Mkdir(testCaseOutputBaseDir, 0777)
+	require.NoError(t, err, "creating the base test case output directory should produce no error")
 
-	deployerCtx, err := df.NewDeployerCtx(meteringOperatorImageRepo, meteringOperatorImageTag, namespace, testOutputDir, targetPods, spec)
-	assert.NoError(t, err, "creating a new deployer context should produce no error")
-
-	rf, err := deployerCtx.Setup()
-	assert.NoError(t, err, "deploying metering should produce no error")
-
-	defer func() {
-		err := deployerCtx.Teardown(df.CleanupScriptPath)
-		assert.NoError(t, err, "capturing logs and uninstalling metering should produce no error")
-	}()
+	for _, testFunction := range testInstallFunctions {
+		if testFunction.Skip {
+			continue
+		}
```

Comment:
We can simply skip the loop iteration in the case where we want to skip a test function, e.g.
```go
			InstallSubTests: []InstallTestCase{
				{
					Name:     "testReportingProducesData",
					TestFunc: testReportingProducesData,
					Skip: true,
				},
				{
					Name:     "testReportingProducesCorrectDataForInput",
					TestFunc: testReportingProducesCorrectDataForInput,
					Skip: true,				
				},
			},
```

In the current implementation, I'm not sure how useful this would be but in the future, we could control this via a go test flag for specific tests:
```go
skipTestReportingProducesData := os.Getenv("DISABLE_TEST_REPORTING_PRODUCES_DATA")
...
{
	Name:     "testReportingProducesData",
	TestFunc: testReportingProducesData,
	Skip: skipTestReportingProducesData,
}
```

It may be more useful to be able to skip certain `MeteringConfig` configurations, like installing metering using s3 for storage, than certain, specific test functions.

Reply:
Yeah, there's no reason to skip individual tests like this except for when it's about being able to run the tests, rather than wanting to. For example, like using S3, it's disabled if there's no creds for it, or something.

Reply:
Alright, removed.